From: Benjamin Kaduk <kaduk@mit.edu>
Date: Mon, 8 Sep 2014 13:47:33 -0400
Subject: Tweak AFSDIR_PATH_MAX definition

On recent Debian, we run into runtime errors in the test suite
because _POSIX_PATH_MAX is only 256, and that buffer is too small
for a call to realpath().  Use PATH_MAX if it's available and larger
than _POSIX_PATH_MAX, in a way that should be safe even when PATH_MAX
is not defined.

Change-Id: I39127e88d92b358245ece21131219380ca4be98a
(cherry picked from commit a16bd1511d47089f9852015331525eab19634668)
---
 src/util/dirpath.hin | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/src/util/dirpath.hin b/src/util/dirpath.hin
index 9ffd341..6fd901f 100644
--- a/src/util/dirpath.hin
+++ b/src/util/dirpath.hin
@@ -75,7 +75,14 @@
 #ifdef AFS_NT40_ENV
 #define AFSDIR_PATH_MAX MAX_PATH
 #else /* unices */
-#define AFSDIR_PATH_MAX    _POSIX_PATH_MAX
+# ifndef PATH_MAX
+#  define PATH_MAX          1024
+# endif
+# if PATH_MAX > _POSIX_PATH_MAX
+#  define AFSDIR_PATH_MAX   PATH_MAX
+# else
+#  define AFSDIR_PATH_MAX   _POSIX_PATH_MAX
+# endif
 #endif
  
 
